src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -296,91 +286,141 @@ runTestCase( const std::shared_ptr<xrt_core::device>& _dev, const std::string& p
+    switch (mode) {
+      case test_mode::json:
+        exit_code = XBU::runScript("sh", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        break;
+      case test_mode::support:
+        exit_code = XBU::runScript("sh", xrtTestCasePath, args, "Checking Test Support", "Test Support Check Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        break;
+      case test_mode::no_json:
+        if (_ptTest.get<std::string>("file").find(".exe") != std::string::npos)
+          exit_code = XBU::runScript("", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        else
+          exit_code = XBU::runScript("python", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
+        break;
+      default:
+        return false; //Mode was not specified correctly.
+    }
 
-    std::vector<std::string> args = { "-p", test_dir,
-                                      "-d", xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(_dev)) };
-    try {
-      int exit_code = XBU::runScript("sh", xrtTestCasePath, args, "Running Test", "Test Duration", MAX_TEST_DURATION, os_stdout, os_stderr, true);
-      if (exit_code == EOPNOTSUPP) {
-        _ptTest.put("status", test_token_skipped);
-      }
-      else if (exit_code == EXIT_SUCCESS) {
+    if (exit_code == EOPNOTSUPP) {
+      logger(_ptTest, "Details", "Test is not supported on current device.");
+      _ptTest.put("status", test_token_skipped);
+      return false;
+    } else if (exit_code == EXIT_SUCCESS) {
+      if (mode != test_mode::support)
         _ptTest.put("status", test_token_passed);
-      }
-      else {
-        logger(_ptTest, "Error", os_stdout.str());
-        logger(_ptTest, "Error", os_stderr.str());
-        _ptTest.put("status", test_token_failed);
-      }
-    } catch (const std::exception& e) {
-      logger(_ptTest, "Error", e.what());
+      return true;
+    } else {
+      logger(_ptTest, "Error", os_stdout.str());
+      logger(_ptTest, "Error", os_stderr.str());
       _ptTest.put("status", test_token_failed);
+      return false;
     }
+  } catch (const std::exception& e) {
+    logger(_ptTest, "Error", e.what());
+    _ptTest.put("status", test_token_failed);
+    return false;
   }
-  else {
-    //check if testcase is present
-    std::string xrtTestCasePath = "/opt/xilinx/xrt/test/" + py;
-    boost::filesystem::path xrt_path(xrtTestCasePath);
-    if (!boost::filesystem::exists(xrt_path)) {
-      logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
-      logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
-      _ptTest.put("status", test_token_failed);
-      return;
-    }
-    // log testcase path for debugging purposes
-    logger(_ptTest, "Testcase", xrtTestCasePath);
+}
```

Comment:
There is no return if the code makes it to the end of the function. I think it should return false if we get here.